src/DependencyInjection/Configuration.php

```diff
@@ -14,10 +14,10 @@
 
     public function getConfigTreeBuilder() : TreeBuilder
     {
-        $treeBuilder = new TreeBuilder();
+        $treeBuilder = new TreeBuilder(KafkaExtension::ALIAS);
 
         /** @var ArrayNodeDefinition $rootNode */
-        $rootNode = $treeBuilder->root(KafkaExtension::ALIAS);
+        $rootNode = method_exists(TreeBuilder::class, 'getRootNode') ? $treeBuilder->getRootNode() : $treeBuilder->root(KafkaExtension::ALIAS);
```

Comment:
Can you make it branched? One for older symfony versions and one for new ones? Also constructor changed so it will be cleaner.

Reply:
Looking at composer.json restrictions everything will work. In popular bundles they don't split repo. Check MonologBundle or DoctrineBundle.

Reply:
I meant eg. do not construct `TreeBuilder` like this `$treeBuilder = new TreeBuilder(KafkaExtension::ALIAS);` because it's incorrect in Symfony 4.2. That needs to be handled then.

Correct way is ``$treeBuilder = new TreeBuilder();`

Reply:
Actually it's correct:

```
 public function __construct(string $name = null, string $type = 'array', NodeBuilder $builder = null)
    {
        if (null === $name) {
            @trigger_error('A tree builder without a root node is deprecated since Symfony 4.2 and will not be supported anymore in 5.0.', E_USER_DEPRECATED);
        } else {
            $this->root($name, $type, $builder);
        }
    }
```

Earlier there was no constructor so creating `new TreeBuilder()` was correct
https://github.com/symfony/symfony/blob/3.0/src/Symfony/Component/Config/Definition/Builder/TreeBuilder.php

Reply:
Oh I see, this is correct then. 👍